plugins/woocommerce/includes/emails/class-wc-email.php

```diff
@@ -801,57 +801,86 @@ public function get_content() {
-
 			/**
-			 * Provides an opportunity to filter the CSS styles included in e-mails.
+			 * Filter to allow the ability to override the email inline styling method.
 			 *
-			 * @since 2.3.0
+			 * @since 10.2.0
 			 *
-			 * @param string    $css   CSS code.
-			 * @param \WC_Email $email E-mail instance.
+			 * @param callable $style_inline_callback The default email inline styling callback.
+			 * @param string|null $content Content that will receive inline styles.
+			 * @param WC_Email $this The WC_Email object.
 			 */
-			$css = apply_filters( 'woocommerce_email_styles', $css, $this );
+			$style_inline_callback = apply_filters( 'woocommerce_mail_style_inline_callback', array( $this, 'apply_inline_style' ), $content, $this );
 
-			$css_inliner_class = CssInliner::class;
+			if ( ! is_callable( $style_inline_callback ) ) {
+				$style_inline_callback = array( $this, 'apply_inline_style' );
+			}
 
-			if ( $this->supports_emogrifier() && class_exists( $css_inliner_class ) ) {
-				try {
-					$css_inliner = CssInliner::fromHtml( $content )->inlineCss( $css );
+			return call_user_func( $style_inline_callback, $content );
+		}
 
-					do_action( 'woocommerce_emogrifier', $css_inliner, $this );
+		return $content;
+	}
 
-					$dom_document = $css_inliner->getDomDocument();
 
-					// When the email is rendered in the block editor, we don't want to remove the elements with display: none.
-					// The main reason is using preview text in the email body which is hidden by default.
-					if ( ! $this->block_email_editor_enabled ) {
-						HtmlPruner::fromDomDocument( $dom_document )->removeElementsWithDisplayNone();
-					}
-					$content = CssToAttributeConverter::fromDomDocument( $dom_document )
-						->convertCssToVisualAttributes()
-						->render();
-				} catch ( Exception $e ) {
-					$logger = wc_get_logger();
-					$logger->error( $e->getMessage(), array( 'source' => 'emogrifier' ) );
+	/**
+	 * Apply inline styles to dynamic content using Emogrifier library (if supported).
+	 *
+	 * @since 10.2.0
+	 * @param string|null $content Content that will receive inline styles.
+	 * @return string
+	 */
+	private function apply_inline_style( $content ) {
+		$css  = '';
+		$css .= $this->get_must_use_css_styles();
+		$css .= "\n";
+
+		ob_start();
+		wc_get_template( 'emails/email-styles.php' );
+		$css .= ob_get_clean();
+
+		/**
+		 * Provides an opportunity to filter the CSS styles included in e-mails.
+		 *
+		 * @since 2.3.0
+		 *
+		 * @param string    $css   CSS code.
+		 * @param \WC_Email $email E-mail instance.
+		 */
+		$css = apply_filters( 'woocommerce_email_styles', $css, $this );
+
+		$css_inliner_class = CssInliner::class;
+
+		if ( $this->supports_emogrifier() && class_exists( $css_inliner_class ) ) {
+			try {
+				$css_inliner = CssInliner::fromHtml( $content )->inlineCss( $css );
+
+				do_action( 'woocommerce_emogrifier', $css_inliner, $this ); // phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment
```

Comment:
Thank you once again.
You're getting the lint errors because you updated (touched) the filters, and the lint rule requires a doc comment for all filters.

Please add the doc comment here as well.

* `@since` 4.1.0